packages/components/CONTRIBUTING.md

```diff
@@ -236,6 +232,74 @@ A couple of good examples of how hooks are used for composition are:
 
 TDB -->
 
+## Naming Conventions
+
+It is recommended for compound components to use dot notation to separate the namespace from the individual component names. The top-level compound component should be called as the namespace (no dot notation).
+
+Dedicated React context should also use the dot notation, while hooks should not.
+
+```tsx
+// Component.tsx
+//=======================
+import { forwardRef, createContext } from '@wordpress/element';
+
+function TopLevelComponent = forwardRef( function TopLevel( props, ref ) { /* ... */ });
+
+function SubComponent = forwardRef( function Sub( props, ref ) { /* ... */ });
+SubComponent.displayName = 'Component.SubComponent';
+
+const Context = createContext();
+
+/** The top-level component's JSDoc. */
+export const Component = Object.assign(
```

Comment:
Could be a good opportunity to explain the purpose behind using `Object.assign()` here.

Reply:
Yes, and it might be useful to state what exactly we need to look for:

- When consuming the component/subcomponent, the JSDocs must appear correctly in IntelliSense.
- The main component JSDoc must appear in the Storybook docs page.
- The component/subcomponent prop types must appear correctly in the Storybook props table.

Reply:
Added a couple of paragraphs explaining requirements and recommended code practices to meet those requirements

https://github.com/WordPress/gutenberg/pull/63714/commits/432635b3f1242b0d6a3b95e54e9d8d36af0b5519